src/classes/ValidatedTopic.ts

```diff
@@ -504,6 +521,68 @@ export default class ValidatedTopic {
     return this.getState().archivist.del(this.getTopic(), this.getIndex())
   }
 
+  /**
+   * Check if the current topic is locked.
+   */
+  public async isLocked(state: mage.core.IState = new mage.core.State()) {
```

Comment:
Not sure about TypeScript, but in ES (afaik) if you return a Promise, you should not mark your function `async`.

Reply:
It doesn't matter: if you return a promise on an async call, then that promise will be the promise to be returned.

`tslint` is currently configured to request that functions returning promises or otherwise using `await` require the `async` keyword. I believe this is for readability purposes (e.g. when checking the function signature), but also to make sure that whatever you return *is* a promise.

Reply:
Do you know if that is true for ES as well?

Reply:
Yes it is. We already had this discussion :)